plugins/woocommerce/src/Utilities/OrderUtil.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * A class of utilities for dealing with orders.
+ */
+
+namespace Automattic\WooCommerce\Utilities;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+use WC_Order;
+use WP_Post;
+
+/**
+ * A class of utilities for dealing with orders.
+ */
+final class OrderUtil {
```

Comment:
In this case, there is no shared state as well. If you are referring to the global `theorder` then see my next comment.

Reply:
I meant "shared state" in a generic sense, as in "anything that is not strictly the arguments passed to the method". In this case, the result of the methods of the class depend on the database state in various ways.

Let me illustrate my concern about testability with an example. Imagine that we are testing a second class that calls `OrderUtil::get_post_or_object_meta($some_post)`, and we want to verify that the tested class behaves differently depending on whether the passed post has a given meta key or not:

* If `get_post_or_object_meta` is static, in order to setup the test we need to either actually introduce some meta values in the database (cumbersome) or patch the static method with the code hacker (not ideal, this should only be needed for legacy code).
* But if `get_post_or_object_meta` is a regular instance method, and the tested class does `wc_get_container()->get(OrderUtil::class)->get_post_or_object_meta($some_post)` instead, then in the test we can easily replace the instance of `OrderUtil` with a mock that directly returns whatever we want (especially since we introduced the `DynamicDecorator` class).

I realize that static methods are more convenient to use, and especially for external contributors being able to just e.g. `OrderUtil::custom_orders_table_usage_is_enabled()` would be very conveniet. But this is something to be considered and solved separately (e.g. by adding a second static version of the method, or adding a proxy class with static methods of the class, or something of that style).

Reply:
I see thanks, I have updated by adding `instance` method of these functions in https://github.com/woocommerce/woocommerce/pull/33638/commits/a8d4a7f8bf8be97fc33fccc1ae4d2df23cd6607f. Note that `custom_orders_table_usage_is_enabled` is not `instansicized` as it is already a static proxy for an instance method.